typing/typecore.ml

```diff
@@ -780,6 +794,10 @@ module Label = NameChoice (struct
     | Record_unboxed true | Record_inlined _ | Record_extension _ -> false
 end)
 
+(* In record-construction expressions and patterns, we have many labels
+   at once; find a candidate type in the intersection of the candidates
+   of each label. In the [closed] expression case, this candidate must
+   contain exactly all the labels. *)
 let disambiguate_label_by_ids keep closed ids labels =
```

Comment:
The explanation makes sense, and I'm fine with merging it as is.
But an explanation of `keep` as well as of the returned boolean would have been nice.

Reply:
I just had a look. I think that it is covered by the following (existing) comment at the callsite:

```
       * if there is no known type reduce it incrementally, so that
         there is still at least one candidate (for error message)
```

The code proceeds by successive refinements/filterings of a list of "potential candidates". If I understand correctly, the idea is to stop reducing the set of possible candidates just before it becomes empty, and use the remaining candidates for suggestions in error messages. `keep` thus means something like "keep the result of the last filtering before the failure", and the boolean indicates whether it was a success. This could be usefully rewritten as a `(labels, labels) result`, with a comment that the Error case returns the last-nonempty list.

Reply:
Thanks!
I think you're that using a `(labels, labels) result` would be nicer.

Reply:
```ocaml
  match List.filter check_ids labels with
    | [] -> Error labels
    | labels ->
  match List.filter check_closed labels with
    | [] -> Error labels
    | labels ->
  Ok labels
```

Reply:
-1 to weird indentation style. If you're going to start suggesting this weird style in all you PRs please start a discussion on caml-devel about whether it is acceptable. Otherwise we're going to have this discussion on all of your PRs individually.